gdnative-core/src/core_types/variant.rs

```diff
@@ -1881,6 +1882,514 @@ macro_rules! impl_variant_for_tuples {
 
 impl_variant_for_tuples!(T1, T2, T3, T4, T5, T6, T7, T8, T9, T10, T11, T12,);
 
+#[cfg(feature = "serde")]
+pub mod serde {
+    use super::super::{
+        dictionary::serde::DictionaryVisitor, variant_array::serde::VariantArrayVisitor,
+    };
+    use super::*;
+    use ::serde::{
+        de::{EnumAccess, Error, MapAccess, SeqAccess, VariantAccess, Visitor},
+        ser::SerializeMap,
+        Deserialize, Deserializer, Serialize, Serializer,
+    };
+    use std::fmt::Formatter;
+
+    impl Serialize for Variant {
+        #[inline]
+        fn serialize<S>(&self, ser: S) -> Result<<S as Serializer>::Ok, <S as Serializer>::Error>
+        where
+            S: Serializer,
+        {
+            match self.dispatch() {
+                VariantDispatch::Nil => ser.serialize_none(),
+                VariantDispatch::Bool(v) => ser.serialize_bool(v),
+                VariantDispatch::I64(v) => ser.serialize_i64(v),
+                VariantDispatch::F64(v) => ser.serialize_f64(v),
+                VariantDispatch::GodotString(v) => ser.serialize_str(&v.to_string()),
+                VariantDispatch::Vector2(v) => {
+                    let mut ser = ser.serialize_map(Some(1))?;
+                    ser.serialize_entry(&VariantType::Vector2, &v)?;
+                    ser.end()
```

Comment:
Consider using [`serialize_newtype_variant`](https://docs.serde.rs/serde/trait.Serializer.html#tymethod.serialize_newtype_variant) which is specifically for externally-tagged enums. This would perform better with non-self-describing (binary) formats (e.g. Bincode) since they won't have to serialize the keys as strings.

Reply:
I don't think non-self-describing formats can directly deserialize `Variant` due to the use of `deserialize_any()` anyway... But they should be able to [de]serialize them as `VariantDispatch`, and as I said in the comment below, my most recent changes should allow most self-describing formats to deserialize a `VariantDispatch` directly as a `Variant` anyway.

Reply:
I see. Perhaps this is worth noting in the type-level docs for `Variant`?